where-to-travel/src/main/webapp/script.js

```diff
@@ -226,59 +287,67 @@ function addPlacesFromDirection(lat, lng, place_candidates) {
   });
 }
 
-/** 
+/**
  * Filter through tourist attractions to find which are in the given time frame of the user.
  *
  * @param {number} time How much time the user wants to travel for
  * @param {array} listPlaces Array of place objects
  * @return {array} An array of places objects that are in the given time frame
  */
- function filterByDistance(timeObj, listPlaces) {
-  return new Promise(function(resolve) {
-    var userLocation = new google.maps.LatLng(home.lat, home.lng);
-    var userDestinations = [];
-    var acceptablePlaces = [];
+function filterByDistance(timeObj, listPlaces) {
+  return new Promise(function (resolve) {
+    const userLocation = new google.maps.LatLng(home.lat, home.lng);
+    let userDestinations = [];
+    let acceptablePlaces = [];
 
     const time = timeObj.hours * 3600 + timeObj.minutes * 60;
-    
-    //iterate through listPlaces and to get all the destinations
-    for (var i = 0; i < 25; i++) {
+
+    // iterate through listPlaces and to get all the destinations
+    let i;
+    for (i = 0; i < 25; i++) {
       const lat = listPlaces[i].geometry.location.lat();
       const lng = listPlaces[i].geometry.location.lng();
-      let destination = new google.maps.LatLng(lat, lng);
+      const destination = new google.maps.LatLng(lat, lng);
       userDestinations.push(destination);
     }
 
-    var service = new google.maps.DistanceMatrixService();
-    service.getDistanceMatrix({
-      origins: [userLocation],
-      destinations: userDestinations,
-      travelMode: 'DRIVING',
-      unitSystem: google.maps.UnitSystem.IMPERIAL,
-    }, callback);
+    let service = new google.maps.DistanceMatrixService();
+    service.getDistanceMatrix(
+      {
+        origins: [userLocation],
+        destinations: userDestinations,
+        travelMode: 'DRIVING',
+        unitSystem: google.maps.UnitSystem.IMPERIAL,
+      },
+      callback
+    );
 
-    function callback(response, status) {    
+    function callback(response, status) {
       if (status == 'OK') {
-        var origins = response.originAddresses;
-        var destinations = response.destinationAddresses;
-      
-        for (var i = 0; i < origins.length; i++) {
-          var results = response.rows[i].elements;
-          for (var j = 0; j < results.length; j++) {
-            var element = results[j];
-            //Check if the time is within the +- 30 min = 1800 sec range
-            if (element.duration.value < time + 1800 && element.duration.value > time - 1800) {
-              acceptablePlaces.push({
-                "name": listPlaces[j].name,
-                "geometry" : listPlaces[j].geometry,
-                "timeInSeconds": element.duration.value,
-                "timeAsString": element.duration.text
-              });
+        const origins = response.originAddresses;
+
+        let i;
+        for (i = 0; i < origins.length; i++) {
```

Comment:
Does it make sense to iterate through this if we always have one origin? If somehow the response had multiple rows, acceptablePlaces will have some junk information. Maybe it is worth adding a check instead that there is only 1 row.